### PR TITLE
solution + test for 109-115

### PR DESCRIPTION
**Pull request by IvanMenov**

### CODE REVIEW

File: src/main/java/g0101_0200/s0110_balanced_binary_tree/Solution.java [hunk 1]
```diff
@@ -4,10 +4,10 @@
 
 public class Solution {
     public boolean isBalanced(TreeNode root) {
-
+        boolean isBalanced = false;
         // Empty Tree is balanced
         if (root == null) {
-            return true;
+            isBalanced = true;
```
Comment (javadev) [inline src/main/java/g0101_0200/s0110_balanced_binary_tree/Solution.java:10]: It should be return here, not assignment value for variable.

File: src/main/java/g0101_0200/s0115_distinct_subsequences/Solution.java [hunk 2]
```diff
@@ -2,45 +2,49 @@
 
 public class Solution {
     public int numDistinct(String text, String text2) {
+        int result;
         if (text.length() < text2.length()) {
-            return 0;
+            result = 0;
```
Comment (javadev) [inline src/main/java/g0101_0200/s0115_distinct_subsequences/Solution.java:7]: It should be return here, not assignment value for variable.

**Review by javadev — CHANGES_REQUESTED** [review]

File: src/main/java/g0101_0200/s0110_balanced_binary_tree/Solution.java [hunk 3]
```diff
@@ -31,13 +31,14 @@ public boolean isBalanced(TreeNode root) {
...
             if (isBalanced(root.left) && isBalanced(root.right)) {
-                return true;
+                isBalanced = true;
             } else {
-                return false;
+                isBalanced = false;
             }
         } else {
-            return false;
+            isBalanced = false;
```
Comment (javadev) [inline src/main/java/g0101_0200/s0110_balanced_binary_tree/Solution.java]: Will it work in leetcode?

File: src/main/java/g0101_0200/s0115_distinct_subsequences/Solution.java [hunk 4]
```diff
@@ -2,45 +2,49 @@
...
+                    }
+                }
+                // No match found for current character of t in s. No point in checking others.
+                if (dp[move - 1] == 0) {
+                    result = 0;
                 }
-            }
-            // No match found for current character of t in s. No point in checking others.
-            if (dp[move - 1] == 0) {
-                return 0;
```
Comment (javadev) [inline src/main/java/g0101_0200/s0115_distinct_subsequences/Solution.java]: Why return was removed? Will it work in leetcode?

**Review by javadev — CHANGES_REQUESTED** [review]

File: src/main/java/g0101_0200/s0110_balanced_binary_tree/Solution.java (hunk 3 above)
Comment (IvanMenov) [reply src/main/java/g0101_0200/s0110_balanced_binary_tree/Solution.java]: it worked

**Comment by javadev** [comment]
I found one code smell.

image

**Pull request merged**